application/models/services/UploadValidator.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+namespace LimeSurvey\Models\Services;
+
+class UploadValidator
+{
+
+    /** @var array<string,mixed> HTTP POST variables*/
+    private $post;
+
+    /** @var array<string,mixed> HTTP File Upload variables*/
+    private $files;
+
+    /**
+     * UploadValidator constructor.
+     *
+     * @param array<string,mixed>|null  $post   HTTP POST variables. If null, $_POST is used.
+     * @param array<string,mixed>|null  $post   HTTP File Upload variables. If null, $_FILES is used.
+     */
+    public function __construct($post = null, $files = null)
+    {
+        $this->post = is_null($post) ? $_POST : $post;
```

Comment:
Please remove this shortcut. ;) The dependency on super-globals should be explicit.

Reply:
Same with line below.

Reply:
Ease of reading > ease of use.

Reply:
Let me read again

Reply:
Maybe you want to inject this instead?
```Yii::app()->request```

Reply:
I like it

Reply:
> Maybe you want to inject this instead?
> `Yii::app()->request`

I'm not sure you can get $_POST array via `Yii::app()->request` …

Reply:
No : https://www.yiiframework.com/doc/api/1.1/CHttpRequest#getPost-detail

`$name` param is mandatory

Reply:
PS : I already check if yii have something ;) .

And same : Yii (1) don't have setPost or setParam :)

Reply:
Good! $_POST and $_GET should be read-only. ^^

Reply:
Yes, imagine when we have unit-tests for all those functions, Denis. Phew. Champagne!

Reply:
> How ! Need to hire hundred of developer ! rofl

Each big method could be a separate class (command object pattern). We inject all dependencies. Then we take it one method at a time.

Reply:
> @gabrieljenik Maybe I can just merge and apply the changes I wanted. ^^

Go for it! Sure!